src/Compatibility/Microsoft.DotNet.PackageValidation/ApiCompatRunner.cs

```diff
@@ -80,6 +82,7 @@ public void RunApiCompat()
                             difference.DiagnosticId,
                             difference.Message);
                     }
+                    counter++;
```

Comment:
nit: you can inline this to the usage of the variable as you only use it once inside the while. 

```cs
(MetadataInformation rightAssembly, string header) rightTuple = _dict[left][counter++];
```